### DISCUSSION TITLE
Shifting between windows changing size but not position.

### DISCUSSION THREAD

**Discussion by adudenamedruby** [ianyh/Amethyst#1510] [Q&A]
Here's the behaviour that I'm after, and I'm curious if Amethyst can do this?

Say I have two windows open: Notes and Mail
- The currently selected window (say Mail) should be 3/4 of the screen and on the left, and the other (Notes) should be 1/4 and on the right.
- If I switch to Notes, it becomes the focus window, but remains on the right, and it changes from being 1/4 of the screen, to being 3/4 of the screen.

Is this possible?

**Comment by ianyh** [comment]
You could accomplish this with a custom layout. You can keep track of the focused window and update its position to be visible with whatever size you want to set.

**Reply by adudenamedruby** [reply]
Ooooo! Interesting. Tyvm!

**Reply by adudenamedruby** [reply]
Heya. I finally got around to starting to put this together. I'm a dev, so I'm able to make some progress here kinda figuring things out step by step (using the docs and the example custom layouts), but I'm very unfamiliar with this stuff (and also don't write JS at all) so it's been slow going.

An idea that might be really helpful for situations like this (and also may improve the app structure overall) that I thought of during the process of putting this together, is to not have built-in layouts defined in the project in swift, but actually use the custom layout format for the default layouts as well. This would have an upside of people being able to see what the defaults layouts look like in JS allowing for better custom layouts, and, also, you'd then only have to maintain a single layout format to deal with, which becomes a win for everyone.

Anyway. Also super happy to see that Amethyst looks in the `.config` for a `amethyst.yml` file.

Great job and a fantastic app! :D
